src/utils/cache.rs

```diff
@@ -1,19 +1,19 @@
-use anyhow::Result;
 use rocket::response::content;
 use rocket_contrib::databases::redis::{self, pipe, Commands, Iter, PipelineCommands};
 use serde::ser::Serialize;
 use serde_json;
 use mockall::automock;
+use crate::utils::errors::{ApiResult, ApiError, BackendError, ApiErrorMessage};
 
 #[database("service_cache")]
 pub struct ServiceCache(redis::Connection);
 
 #[automock]
 pub trait Cache {
     fn fetch(&self, id: &str) -> Option<String>;
-    fn create(&self, id: &String, dest: &String, timeout: usize);
-    fn invalidate_pattern(&self, pattern: &String);
-    fn invalidate(&self, id: &String);
+    fn create(&self, id: &str, dest: &str, timeout: usize);
```

Comment:
Why switch to `str` with this PR? Can we use `impl Into<String>` then it can handle `str` and `String`

Reply:
I think it's generally better to use `&str` (string slice) as it's immutable and generally we are just borrowing the value: https://stackoverflow.com/questions/24158114/what-are-the-differences-between-rusts-string-and-str#:~:text=A%20Rust%20String%20is%20like,contents%20of%20std%3A%3Astring%20.